### PR TITLE
Add CLI option to force output resample rate

### PR DESCRIPTION
**Pull request by christophehenry**

### CODE REVIEW

**Comment by sezanzeb** [comment]
Can't run the ui anymore:

```
➜  soundconverter git:(resample-rate) soundconverter
/usr/lib/python3.9/site-packages/soundconverter/interface/preferences.py:27: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '4.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk, GLib
/usr/lib/python3.9/site-packages/soundconverter/gstreamer/converter.py:26: PyGIWarning: Gst was imported without specifying a version first. Use gi.require_version('Gst', '1.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gst, GLib, Gio
ERROR: Disabling flac-enc output. Do you have "gst-plugins-good" installed?
ERROR: Disabling wav-enc output. Do you have "gst-plugins-good" installed?
ERROR: Disabling vorbis-enc output. Do you have "gst-plugins-base" installed?
ERROR: Disabling vorbis-mux output. Do you have "gst-plugins-base" installed?
ERROR: Disabling mp3-id-tags output. Do you have "gst-plugins-good" installed?
ERROR: Disabling mp3-vbr-tags output. Do you have "gst-plugins-ugly" installed?
ERROR: Disabling mp3-enc output. Do you have "gst-plugins-good" installed?
ERROR: Disabling aac-mux output. Do you have "gst-plugins-good" installed?
ERROR: Disabling opus-enc output. Do you have "gst-plugins-base" installed?
ERROR: Disabling aac-enc output. Do you have "gst-plugins-bad" installed?
soundconverter needs GTK >= 3.0 (Error: "Namespace Gtk is already loaded with version 4.0")
```

**Comment by kassoulet** [comment]
Hi Christophe,

I'm afraid tests are not passing. I'll look when I get some time, but feel
free to resubmit if you manage to fix them.

======================================================================
ERROR: test_validate_args (testcases.batch.BatchUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File
"/home/gautier/Downloads/soundconverter-resample-rate/tests/testcases/batch.py",
line 268, in test_validate_args
    self.assertFalse(validate_args({'output-resample': 160}))
  File
"/usr/local/lib/python3.8/dist-packages/soundconverter/interface/batch.py",
line 243, in validate_args
    "Possible values for -R or --output-resample option are
{rates}".format(rates=", ".join(rates))
TypeError: sequence item 0: expected str instance, int found

======================================================================
FAIL: test_set_output_sample (testcases.batch.BatchUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File
"/home/gautier/Downloads/soundconverter-resample-rate/tests/testcases/batch.py",
line 370, in test_set_output_sample
    self.assertEqual(44100, gio_settings.get_boolean('output-resample'))
AssertionError: 44100 != True

======================================================================
FAIL: test_set_output_resample (testcases.integration.BatchIntegration)
----------------------------------------------------------------------
Traceback (most recent call last):
  File
"/home/gautier/Downloads/soundconverter-resample-rate/tests/testcases/integration.py",
line 464, in test_set_output_resample
    self.assertIsNone(gio_settings.get_boolean('output-resample'))
AssertionError: False is not None

======================================================================
FAIL: test_conversion_pattern (testcases.integration.GUI)
----------------------------------------------------------------------
Traceback (most recent call last):
  File
"/home/gautier/Downloads/soundconverter-resample-rate/tests/testcases/integration.py",
line 911, in test_conversion_pattern
    self.assertTrue(os.path.isfile(
AssertionError: False is not true

File: tests/testcases/integration.py [hunk 1]
```diff
@@ -452,6 +452,26 @@ def test_set_delete_original_true(self):
...
+            '-b', 'tests/tmp/a.wav',
+            '-o', 'tests/tmp',
+            '-f', 'm4a',
+            '-R', '8000'
+        ])
+
+        gio_settings = get_gio_settings()
+        self.assertTrue(gio_settings.get_boolean('output-resample'))
+        self.assertEqual(gio_settings.get_int('resample-rate'), 8000)
+        self.assertEqual(self.get_bitrate('tests/tmp/a.m4a'), 8000)
```
Comment (christophehenry) [inline tests/testcases/integration.py:473]: This test is failing. I'm not sure how `get_bitrate()` computes bitrate.

**Comment by christophehenry** [comment]
@sezanzeb Fixed.
@kassoulet Fixed too. Only one test failing because I don't know how `get_bitrate()` actually computes the bitrate. If you could help me.

File: tests/testcases/integration.py (hunk 1 above)
Comment (sezanzeb) [reply tests/testcases/integration.py:473]: should be close to file-size / duration.

if you use wav in this test, the bitrate might be quite close to the samplerate.

File: tests/testcases/integration.py [hunk 2]
```diff
@@ -452,6 +452,28 @@ def test_set_delete_original_true(self):
...
+            '-b', 'tests/tmp/c.mp3',
+            '-o', 'tests/tmp',
+            '-f', 'wav',
+            '-R', str(sample_rate)
+        ])
+
+        gio_settings = get_gio_settings()
+        self.assertTrue(gio_settings.get_boolean('output-resample'))
+        self.assertEqual(gio_settings.get_int('resample-rate'), sample_rate)
+        self.assertEqual(self.get_bitrate('tests/tmp/c.wav'), sample_rate * 8 / 1000 * 2)
```
Comment (christophehenry) [inline tests/testcases/integration.py:475]: This formula looks correct. I took inspiration from [this test](https://github.com/kassoulet/soundconverter/blob/main/tests/testcases/integration.py#L243-L265).

**Comment by kassoulet** [comment]
Nice, thank you Christophe !

**Pull request merged**